contrib/babelfishpg_tsql/src/pl_exec.c

```diff
@@ -10045,6 +10045,18 @@ bool reset_search_path(PLtsql_stmt_execsql *stmt, char *old_search_path)
 	const char	*user = NULL;
 	const char	*schema;
 	top_es_entry = exec_state_call_stack->next;
+
+	if (stmt->func_call && stmt->schema_name != NULL && strncmp(stmt->schema_name, "sys", 3) != 0)
+	{
+		physical_schema = get_physical_schema_name(cur_dbname, stmt->schema_name);
```

Comment:
who manages memory for these variables?

Reply:
These are local variables and we are not dynamically allocating any memory, it gets stored in the stack while function execution and gets cleaned up as it goes out of scope. We don't need to explicitly free the memory.

Reply:
These are local variables stored in the stack and gets cleaned up at the end of the function execution. Since we are not allocated the memory dynamically, we don't need to free it explicitly.